pkgs/build-support/rust/build-rust-package/default.nix

```
@@ -108,7 +109,7 @@ lib.extendMkDerivation {
           cargoDeps
         else if cargoLock != null then
           importCargoLock cargoLock
-        else if args.cargoHash or null == null then
+        else if finalAttrs.cargoHash == null then
```

The closest solution to override `cargoHash` at the main package and override the rest of `cargoDeps` attributes at `cargoDeps` is to
- Select the candidate using the initial attribute
- Define the cargo vendor candidate with `finalAttrs.cargoHash`.

```suggestion
        else if cargoHash == null then
```